### ISSUE TITLE
The "itext-demo" demo form fails to load 

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#6706]
**Describe the issue**
This form: https://github.com/medic/support-scripts/blob/54b180613060ce73505bb163886c52f688f47931/forms/itext-demo.xml
fails on `3.10` with 
```
Error loading form. Error: ["FormLogicError: Could not evaluate: jr:itext('/issue-3870/group_note/default_chw_sms/anc_only_home_sba_birth:label'), message: Failed to execute 'evaluate' on 'Document': The string 'jr:itext(\"/issue-3870/group_note/default_chw_sms/anc_only_home_sba_birth:label\")' is not a valid XPath expression."]
```

**Describe the improvement you'd like**
Determine whether this form has ever worked. 
If it has, it means we've regressed at some point. In this case, create a new issue to fix the regression. 
If it's bugged in some minor way, fix it. 
If it's never worked and we're not supposed to support it, remove it from the demo-forms folder. 

For reference, 
- this is the form on `gamma.dev` : https://gamma.dev.medicmobile.org/#/reports/add/issue:3870
- this is the issue that it's referencing https://github.com/medic/cht-core/issues/3870
- the form itself was added in a separate commit one year later: https://github.com/medic/cht-core/commit/c10e08519eeb72e90512e87dd2dcaf3eefed9661

**Comment by dianabarsan** [comment]
Further testing shows that another demo form (https://github.com/medic/cht-core/blob/master/demo-forms/itext-in-calculate.xml) fails with another error, and looks like it's related: 
```
Error loading form. Error: ["FormLogicError: Could not evaluate: jr:itext('/test/one'), message: Failed to execute 'evaluate' on 'Document': The string 'jr:itext(\"/test/one\")' is not a valid XPath expression."]
```

**Comment by MaxDiz** [comment]
@dianabarsan sounds like it should be treated as a regression and scheduled in the next release (or after Angular update is merged in 3.11). Agree?

**Comment by dianabarsan** [comment]
It's unclear if the regression is in CHT-Core or Enketo, and I think it probably happened many versions ago. 
I've found other possible regressions (but not as dramatic as this one) in demo forms. 

I made a note to go over all of them once the Angular migration is done. 

Because we've scheduled the Enketo upgrade for next version (3.12), I'm tempted to push these until after we do the upgrade, as it might fix these issues by itself. They require more investigation, though.

**Comment by dianabarsan** [comment]
I randomly bumped into this. Adding the Enketo tag for investigation, @jkuester .

**Comment by jkuester** [comment]
TLDR: I am planning to address this issue by removing the broken test forms as a part of the Enketo uplift changes here: https://github.com/medic/support-scripts/pull/7

___

I have verified this still does not work on the updated Enketo version in [`6345-enketo-uplift`](https://github.com/medic/cht-core/tree/6345-enketo-uplift).  I also tested against `3.4.0` and also got the same issue. After doing a bunch of reading, I am pretty sure that using `jr:itext` as an xPath function in a `calculate` never worked with Enketo.  As Alex noted in [his comment](https://github.com/medic/cht-core/issues/3870#issuecomment-328077648) on this issue linked above, the `itext` data is unpacked during the xsl transformation of the form xml and is not implemented in such a way that it could be used used dynamically.  

It is also worth noting that `jr:itext` is one of the ODK xPath functions that openrosa-xpath-evaluator explicitly [notes is not implemented](https://github.com/enketo/openrosa-xpath-evaluator/blob/master/README.md#odk-functions).  

These itext test forms were added here in [a commit](https://github.com/medic/cht-core/commit/c10e08519eeb72e90512e87dd2dcaf3eefed9661#diff-35c00d3b3f94849259f8434d846840be2dac3cc4fd5adaf4971d26f584e3add0) that added a bunch of test forms. I am not sure exactly where Alex got those forms, but my guess is at least some of them were for testing ODK Collect (where using `jr:itext` as an xPath function in a `calculate` seems to work, presumably because they have implemented this in javarosa...).

Context:

- https://github.com/medic/cht-core/issues/3870
- https://github.com/enketo/enketo-core/issues/32
- https://github.com/enketo/enketo-xpathjs/issues/43
